packages/effect/src/Effect.ts

```diff
@@ -5984,9 +5984,10 @@ export const Tag: <const Id extends string>(id: Id) => <
   & Context.TagClass<Self, Id, Type>
   & (Type extends Record<PropertyKey, any> ? {
       [
-        k in keyof Type as Type[k] extends ((...args: [...infer Args]) => infer Ret) ?
-          ((...args: Readonly<Args>) => Ret) extends Type[k] ? k : never
-          : k
+        k
+          in keyof Type as (Type[k] extends ((...args: [...infer Args]) => infer Ret) ?
+            ((...args: Readonly<Args>) => Ret) extends Type[k] ? k : never
+            : k) extends infer z extends PropertyKey ? z extends "name" ? never : z : never
```

Comment:
Can we add it to `ProhibitedType`?

Reply:
Or maybe we make the others "passthrough" like `name`.

Reply:
what do you mean by passthrough?

Reply:
ah understood, no I think it is better to prohibit, less risk for confusion